rust/chains/hyperlane-ethereum/src/rpc_clients/fallback.rs

```diff
@@ -98,9 +171,83 @@ impl From<FallbackError> for ProviderError {
     }
 }
 
+async fn handle_stalled_provider<C>(
+    fallback_provider: &FallbackProvider<C>,
```

Comment:
wdyt of putting all these functions in an `impl<C> FallbackProvider where C: JsonRpcClient` so that we can just use `&self`?